discover/deployment/articles/01-installing-liferay/04-installing-liferay-on-tomcat.markdown

```diff
@@ -364,28 +364,28 @@ with the scope set as `compile` (the default) in each JSF project:
     <dependency>
         <groupId>com.sun.faces</groupId>
         <artifactId>jsf-api</artifactId>
-        <version>2.1.21</version>
+        <version>2.1.29-08</version>
         <scope>compile</scope>
     </dependency>
     <dependency>
         <groupId>com.sun.faces</groupId>
         <artifactId>jsf-impl</artifactId>
-        <version>2.1.21</version>
+        <version>2.1.29-08</version>
         <scope>compile</scope>
     </dependency>
 
 Although it is possible to install Mojarra in the Tomcat global classpath, it
 will not work properly without some small modifications to the `jsf-impl.jar`
 dependency. The problem stems from the fact that the Mojarra
-[`ConfigureListener`](https://svn.java.net/svn/mojarra~svn/tags/2.1.21/jsf-ri/src/main/java/com/sun/faces/config/ConfigureListener.java)
+[`ConfigureListener`](https://svn.java.net/svn/mojarra~svn/tags/2.1.29/jsf-ri/src/main/java/com/sun/faces/config/ConfigureListener.java)
 class is automatically registered for all contexts under `tomcat/webapps`
 because it is specified as a `<listener>` in the
-[META-INF/jsf-jsf_core.tld](https://svn.java.net/svn/mojarra~svn/tags/2.1.21/jsf-ri/conf/share/jsf_core.tld)
+[META-INF/jsf-jsf_core.tld](https://svn.java.net/svn/mojarra~svn/tags/2.1.29/jsf-ri/conf/share/jsf_core.tld)
 descriptor inside the `jsf-impl.jar` dependency.
 Additionally, the
-[META-INF/services/javax.faces.ServletContainerInitializer](https://svn.java.net/svn/mojarra~svn/tags/2.1.21/jsf-ri/conf/share/javax.servlet.ServletContainerInitializer)
+[META-INF/services/javax.faces.ServletContainerInitializer](https://svn.java.net/svn/mojarra~svn/tags/2.1.29/jsf-ri/conf/share/javax.servlet.ServletContainerInitializer)
 will cause the
-[FacesInitializer](https://svn.java.net/svn/mojarra~svn/tags/2.1.21/jsf-ri/src/main/java/com/sun/faces/config/FacesInitializer.java)
+[FacesInitializer](https://svn.java.net/svn/mojarra~svn/tags/2.1.29/jsf-ri/src/main/java/com/sun/faces/config/FacesInitializer.java)
```

Comment:
@codyhoag, thanks for updating this so quickly. The latest source code is now in git on github, so you can replace `https://svn.java.net/svn/mojarra~svn/tags/2.1.29` with `https://github.com/javaserverfaces/mojarra/blob/2.1.29-08` and that should work (and it's more up to date). I think that would work for https://github.com/sez11a/liferay-docs/pull/2045 as well.